### PR TITLE
add ide.host.json files to project templates

### PR DESCRIPTION
**Pull request by vlada-shubina**
### Problem
Currently project templates do not have `ide.host.json` file. 
The fallback built-in to Visual Studio is used.

### Solution
Add `ide.host.json` and icons to project templates.

### Checks:
- [ ] Added unit tests
- [ ] Added `#nullable enable` to all the modified files [?](https://github.com/dotnet/templating/wiki/Contributing#coding-style)

### CODE REVIEW

File: template_feed/Microsoft.DotNet.Common.ProjectTemplates.7.0/content/ClassLibrary-CSharp/.template.config/ide.host.json [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+  "$schema": "http://json.schemastore.org/vs-2017.3.host",
+  "icon": "ide/ClassLibraryCS.ico",
+  "order": 20,
+  "tags": [
+    {
+      "type": "projectType",
+      "add": [],
+      "remove": [ "Common" ]
+    }
+  ],
```
Comment (vlada-shubina) [inline template_feed/Microsoft.DotNet.Common.ProjectTemplates.7.0/content/ClassLibrary-CSharp/.template.config/ide.host.json:11]: @baronfel do we need to keep it? would it be better to remove this tag from `template.json`?

File: template_feed/Microsoft.DotNet.Common.ProjectTemplates.7.0/content/ClassLibrary-CSharp/.template.config/ide.host.json [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+  "symbolInfo": [
+    {
+      "id": "langVersion",
+      "isVisible": "true"
```
Comment (vlada-shubina) [inline template_feed/Microsoft.DotNet.Common.ProjectTemplates.7.0/content/ClassLibrary-CSharp/.template.config/ide.host.json:15]: @baronfel this field is text field. Do we need to enable it for VS?

File: template_feed/Microsoft.DotNet.Common.ProjectTemplates.7.0/content/ClassLibrary-CSharp/.template.config/ide.host.json [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+  "icon": "ide/ClassLibraryCS.ico",
+  "order": 20,
+  "tags": [
+    {
+      "type": "projectType",
+      "add": [],
+      "remove": [ "Common" ]
+    }
+  ],
+  "symbolInfo": [
```
Comment (vlada-shubina) [inline template_feed/Microsoft.DotNet.Common.ProjectTemplates.7.0/content/ClassLibrary-CSharp/.template.config/ide.host.json:12]: @sayedihashimi @phenning do we need to enable framework?

Comment (phenning) [reply template_feed/Microsoft.DotNet.Common.ProjectTemplates.7.0/content/ClassLibrary-CSharp/.template.config/ide.host.json:12]: If we don't want/need it in the CLI, I'd actually prefer if we'd remove it from template.json.

**Review by phenning — CHANGES_REQUESTED** [review]
A couple of things.

1. We can probably remove the order from here, since there is a static ordering controlled from VS itself, so this is essentially ignored. 

2. I'd remove skip restore from the symbol info, since VS will always restore. This wouldn't have any effect in VS, since we don't have a Skip restore post action and this would just add confusion if shown in Visual Studio.

3. Language version. If we are going to display this in Visual Studio, we should instead have a seperate choice parameter for it instead as we shouldn't expect customers to know the valid values. We don't have fleshed out validation integration between templates, the host and the Visual Studio new project dialog.

4. if you rename the icons to icon.ico or icon.png and place them in the .template.config or .tempalate.config/ide sub folders, VS will find them without having to specify the icon property in the host file.

File: template_feed/Microsoft.DotNet.Common.ProjectTemplates.7.0/content/ClassLibrary-CSharp/.template.config/ide.host.json (hunk 2 above)
Comment (phenning) [reply template_feed/Microsoft.DotNet.Common.ProjectTemplates.7.0/content/ClassLibrary-CSharp/.template.config/ide.host.json:15]: Comments below, I'd not. If we want control over this, we should add a choice parameter with the valid values.

File: template_feed/Microsoft.DotNet.Common.ProjectTemplates.7.0/content/ClassLibrary-CSharp/.template.config/ide.host.json (hunk 1 above)
Comment (phenning) [reply template_feed/Microsoft.DotNet.Common.ProjectTemplates.7.0/content/ClassLibrary-CSharp/.template.config/ide.host.json:11]: +1 If we don't need it in the tag list:

Class Library                                 classlib             [C#],F#,VB  **Common**/Library

Comment (baronfel) [reply template_feed/Microsoft.DotNet.Common.ProjectTemplates.7.0/content/ClassLibrary-CSharp/.template.config/ide.host.json]: Tags are used in filtering on the CLI and VS, right?  What does "Common" mean here, semantically? If anything it feels like something that's host-specific - move templates used recently to the top of the list, etc.

**Comment by vlada-shubina** [comment]
I also put https://github.com/dotnet/templating/pull/4520 on top of this PR to have `ide.host.json` adapted for `--use-program-main` option.

File: template_feed/Microsoft.DotNet.Common.ProjectTemplates.7.0/content/ClassLibrary-CSharp/.template.config/ide.host.json (hunk 1 above)
Comment (vlada-shubina) [reply template_feed/Microsoft.DotNet.Common.ProjectTemplates.7.0/content/ClassLibrary-CSharp/.template.config/ide.host.json]: in CLI it is possible to filter by tags when doing `list` or `search` using `--tag` option.
Those templates have `Common` tag:
```
PS C:\Users\vshubina> dotnet new list --tag Common
These templates matched your input: --tag='Common'

Template Name                              Short Name           Language    Tags
-----------------------------------------  -------------------  ----------  -----------------
Class Library                              classlib             [C#],F#,VB  Common/Library
Console App                                console              [C#],F#,VB  Common/Console
Web api application using Microsoft Graph  graphwebapi          [C#]        Common/Web
Windows Forms App                          winforms             [C#],VB     Common/WinForms
Windows Forms Class Library                winformslib          [C#],VB     Common/WinForms
Windows Forms Control Library              winformscontrollib   [C#],VB     Common/WinForms
Worker Service                             worker               [C#],F#     Common/Worker/Web
WPF Application                            wpf                  [C#],VB     Common/WPF
WPF Class library                          wpflib               [C#],VB     Common/WPF
WPF Custom Control Library                 wpfcustomcontrollib  [C#],VB     Common/WPF
WPF User Control Library                   wpfusercontrollib    [C#],VB     Common/WPF
```

If you do `dotnet new search --tag Common --columns tags` there are lots of hits for many things.

**Review by phenning — APPROVED** [review]
Program main changes will come in after this one, right?

**Review by baronfel — APPROVED** [review]

**Review by donJoseLuis — APPROVED** [review]

**Pull request merged**